blockchain/types/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Blockchain data structures.
+
+#![no_std]
+#![deny(missing_docs)]
+
+extern crate alloc;
+
+mod block;
+mod block_contents;
+mod block_data;
+mod block_id;
+mod block_signature;
+mod error;
+
+pub use crate::{
+    block::{compute_block_id, Block, BlockIndex, MAX_BLOCK_VERSION},
+    block_contents::{BlockContents, BlockContentsHash},
+    block_data::BlockData,
+    block_id::BlockID,
+    block_signature::BlockSignature,
+    error::ConvertError,
+};
+
+pub use mc_transaction_types::BlockVersion;
```

Comment:
It's originally defined in transaction_core because it's a key flag which defines what kind of operations (like minting, multiple tokens, etc.) can be performed and thus what things to include in the proofs/how the proofs work out (i.e. doing the bulletproofs/range proofs in multiple tokens vs. just MOB). But also Blockversion is used to do logic by multiple actors in the system, so it seems like it could go here just fine and this would be a dependency of core?

Reply:
No, because `mc_blockchain_types` depends on `mc_transaction_core` (mainly for `TxOut`) and `mc_transaction_core` needs `BlockVersion` for a bunch of things. I'm not sure why `BlockVersion` is in `mc_transaction_types`, though.